src/lib/Microsoft.Health.Extensions.Fhir.R4/HttpClientBuilderRegistrationExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Extensions.Host.Auth;
+
+namespace Microsoft.Health.Extensions.Fhir
+{
+    public static class HttpClientBuilderRegistrationExtensions
+    {
+        public static void AddAuthenticationHandler(
+            this IHttpClientBuilder httpClientBuilder,
+            IServiceCollection services,
+            Uri uri,
+            bool useManagedIdentity)
+        {
+            EnsureArg.IsNotNull(httpClientBuilder, nameof(httpClientBuilder));
+            EnsureArg.IsNotNull(services, nameof(services));
+            EnsureArg.IsNotNull(uri, nameof(uri));
+
+            if (useManagedIdentity)
+            {
+                services.AddNamedManagedIdentityCredentialProvider();
+                httpClientBuilder.AddHttpMessageHandler(x =>
+                    ActivatorUtilities.CreateInstance<BearerTokenAuthorizationMessageHandler>(x, new object[] { uri, new ManagedIdentityAuthService() }));
```

Comment:
My initial thinking here was to instantiate the BearerTokenAuthorizationMessageHandler with the logger argument provided directly from the IServiceProvider. However, since we're now configuring and building the logger in AddFhirClient() it makes sense to pass it a new instance of BearerTokenAuthMessageHandler. I've updated the code to reflect that